src/backup.ts

```diff
@@ -0,0 +1,492 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {promisifyAll} from '@google-cloud/promisify';
+import {Instance} from './instance';
+import {
+  IOperation,
+  LongRunningCallback,
+  RequestCallback,
+  ResourceCallback,
+} from './common';
+import {EnumKey, RequestConfig, TranslateEnumKeys} from '.';
+import {Metadata, Operation as GaxOperation} from 'google-gax';
+import {DateStruct, PreciseDate} from '@google-cloud/precise-date';
+import {CallOptions, status} from 'grpc';
+import {google as databaseAdmin} from '../protos/protos';
+
+export type CreateBackupCallback = LongRunningCallback<Backup>;
+
+export interface CreateBackupGaxOperation extends GaxOperation {
+  // Overridden with more specific type for CreateBackup operation
+  metadata: Metadata &
+    databaseAdmin.spanner.admin.database.v1.ICreateBackupMetadata;
+}
+
+export type CreateBackupResponse = [
+  Backup,
+  CreateBackupGaxOperation,
+  IOperation
+];
+
+export interface CreateBackupOptions {
+  databasePath: string;
+  expireTime: PreciseDate;
+  gaxOptions?: CallOptions;
+}
+
+/**
+ * IBackup structure with backup state enum translated to string form.
+ */
+type IBackupTranslatedEnum = TranslateEnumKeys<
+  databaseAdmin.spanner.admin.database.v1.IBackup,
+  'state',
+  typeof databaseAdmin.spanner.admin.database.v1.Backup.State
+>;
+
+export type GetMetadataResponse = [IBackupTranslatedEnum];
+type GetMetadataCallback = RequestCallback<IBackupTranslatedEnum>;
+
+type UpdateExpireTimeCallback = RequestCallback<
+  databaseAdmin.spanner.admin.database.v1.IBackup
+>;
+
+type DeleteCallback = RequestCallback<void>;
+
+interface BackupRequest {
+  (
+    config: RequestConfig,
+    callback: ResourceCallback<GaxOperation, IOperation>
+  ): void;
+  <T>(config: RequestConfig, callback: RequestCallback<T>): void;
+}
+/**
+ * The {@link Backup} class represents a Cloud Spanner backup.
+ *
+ * Create a `Backup` object to interact with or create a Cloud Spanner backup.
+ *
+ * @class
+ *
+ * @example
+ * const {Spanner} = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ * const instance = spanner.instance('my-instance');
+ * const backup = instance.backup('my-backup');
+ */
+class Backup {
+  id: string;
+  formattedName_: string;
+  instanceFormattedName_: string;
+  request: BackupRequest;
+  constructor(instance: Instance, name: string) {
+    this.request = instance.request;
+    this.instanceFormattedName_ = instance.formattedName_;
+    this.formattedName_ = Backup.formatName_(instance.formattedName_, name);
+    this.id = this.formattedName_.split('/').pop() || '';
+  }
+
+  create(options: CreateBackupOptions): Promise<CreateBackupResponse>;
+  create(options: CreateBackupOptions, callback: CreateBackupCallback): void;
+  /**
+   * @typedef {array} CreateBackupOptions
+   * @property {string} 0 The database path.
```

Comment:
I believe these need to be keyed by name, e.g. "{string} databasePath The database path."

Reply:
I've been a bit confused about when to do this. It seems like it only works if you say @typedef {object} so I changed it to that.